### ISSUE TITLE
Camel forge - Limit the number of options per screen

### ISSUE THREAD

**Issue by davsclaus** [fabric8io/fabric8#4405]
If a component has many options the dialog gets too big. We should instead limit to max 20 options per screen and then have additional screens to set all the options.

For testing try eg the jms component which has many options.

**Comment by oscerd** [comment]
Claus, what command do you mean? Camel-info?

Andrea

**Comment by davsclaus** [comment]
Ah its the camel-edit-endpoint or camel-add-endpoint that would have this. For example if you have a Camel project with routes in XML and you have a jms endpoint then the edit dialog shows 80+ options in one page, and therefore it scrolls out of the screen. 

I guess a screenshot would have been good when I logged the ticket.

To reproduce you can load the camel-servlet quickstart into forge
https://github.com/fabric8io/quickstarts/tree/master/quickstarts/war/camel-servlet

And then change the endpoint to jms component or some other component that has many options, eg file or ftp also has many options.

Its likely this for loop that needs to break up into max N per step
https://github.com/fabric8io/fabric8/blob/master/forge/addons/camel/src/main/java/io/fabric8/forge/camel/commands/project/ConfigureEditEndpointPropertiesStep.java#L108

And then have the wizard being a multi step, eg if there is more options, then next is the next group of options, until all of them have been shown.

Maybe the logic in next needs to be dynamic and understand this
https://github.com/fabric8io/fabric8/blob/master/forge/addons/camel/src/main/java/io/fabric8/forge/camel/commands/project/ConfigureEditEndpointPropertiesStep.java#L140

I am not sure how to do this, maybe the for loop can be as is, and you just show/hide N options per step. The forge guys hang out on irc at #forge on freenode and are great at helping out.

**Comment by davsclaus** [comment]
Also maybe this needs to be for the GUI only, as the CLI can just show all the options. 

Also maybe this is a general problem and the forge guys have some thoughts how to make wizards with many options that could auto do multiple steps if needed.

**Comment by oscerd** [comment]
Ok, I'll work on this.

**Comment by oscerd** [comment]
Actually, if I change servlet endpoint to jms endpoint I get something like this in Forge console:

![image](https://cloud.githubusercontent.com/assets/5106647/8765782/a2abd45e-2e1e-11e5-9f05-55675751ea9d.png)

**Comment by oscerd** [comment]
From my point of view the problem is only on the GUI side (I need to do tests). Actually the CLI scale the output based on the options size, so there are no problem (tested with JMS and FTP component)

**Comment by oscerd** [comment]
I'm using Forge-2.17.0.Final

**Comment by davsclaus** [comment]
If you use IDEA then you can install forge plugin from the plugin manager in IDEA. Then it uses the addons you have installed from forge CLI.

You can then start forge from IDEA using ALT + CMD + 4

**Comment by oscerd** [comment]
I'm using Eclipse and, as always, using plugins is a nightmare.
Recently I've done some test with Forge. What I see is the following:

- Some of the new versions of camel, kubernetes, devops, openshift etc. addons are based on the latest version of Forge, 2.16.2 and 2.17.0 (to be precise). Trying to install these addons on older version of Forge doesn't work.
- The latest stable eclipse plugin of Forge uses Forge console 2.15.2.Final, so I'm a little bit in trouble to run tests.

**Comment by davsclaus** [comment]
Yeah for Eclipse its a bit of pain - I think there is some JBDS Eclipse plugins / distribution that has newer Forge version.

**Comment by davsclaus** [comment]
In the upcoming Camel 2.16 we have more labels for each option, so we should group by labels

- consumer
- producer
- advanced
- security
and so on.

**Comment by davsclaus** [comment]
Phew for the first time I got forge to do the multi step wizard based on the number of options to work in CLI and IDEA.

![screen shot 2015-11-10 at 5 25 17 pm](https://cloud.githubusercontent.com/assets/477100/11068010/9f8e7820-87cf-11e5-86bb-dd12b036ad4b.png)

Splitting up the wizard per group is next on the agenda, so we can have common, advanced, security etc separated. And the UI wizard to show that in the title.

The generated code is a bit long as I changed many options

```
	@Inject
	@Uri("jms:myqueue?asyncConsumer=true&defaultTaskExecutorType=ThreadPool&deliveryMode=1&jmsMessageType=Bytes&replyTo=myreply&replyToType=Exclusive&testConnectionOnStartup=true&transferException=true")
	private Endpoint m;
```
